atomic_reactor/plugins/pre_flatpak_create_dockerfile.py

```diff
@@ -135,19 +137,23 @@ def run(self):
 
         # Create the dockerfile
 
-        df_path = os.path.join(self.workflow.df_dir, DOCKERFILE_FILENAME)
-        with open(df_path, 'w') as fp:
-            fp.write(DOCKERFILE_TEMPLATE.format(name=name,
-                                                component=component,
-                                                cleanupscript=FLATPAK_CLEANUPSCRIPT_FILENAME,
-                                                includepkgs=FLATPAK_INCLUDEPKGS_FILENAME,
-                                                stream=module_info.stream.replace('-', '_'),
-                                                base_image=base_image,
-                                                relative_repos_path=RELATIVE_REPOS_PATH,
-                                                rpm_qf_args=rpm_qf_args(),
-                                                yum_repos_dir=YUM_REPOS_DIR))
-
-        self.workflow.set_df_path(df_path)
+        def _create_dockerfile(build_dir: BuildDir) -> List[Path]:
+            content = DOCKERFILE_TEMPLATE.format(name=name,
+                                                 component=component,
+                                                 cleanupscript=FLATPAK_CLEANUPSCRIPT_FILENAME,
+                                                 includepkgs=FLATPAK_INCLUDEPKGS_FILENAME,
+                                                 stream=module_info.stream.replace('-', '_'),
+                                                 base_image=base_image,
+                                                 relative_repos_path=RELATIVE_REPOS_PATH,
+                                                 rpm_qf_args=rpm_qf_args(),
+                                                 yum_repos_dir=YUM_REPOS_DIR)
+            build_dir.dockerfile_path.write_text(content, "utf-8")
+            return [build_dir.dockerfile_path]
+
+        created_files = self.workflow.build_dirs.for_all_copy(_create_dockerfile)
+
+        dockerfile_path = created_files[0]
+        self.workflow.set_df_path(str(dockerfile_path))
```

Comment:
I'm putting the `df_path` property on my "list of things to remove when we're done". For now it has to stay, removing it would break too many things.

Note to self: the method will still be needed, should be renamed to `set_dockerfile_images`

Reply:
In addition, it would be also good to separate the Dockerfile validation from the set operation.